golos.vesting/golos.vesting.cpp

```diff
@@ -447,7 +413,8 @@ const asset vesting::convert_to_token(const asset &m_token, const structures::ve
     if (!vinfo.supply.amount || !this_balance.amount)
         amount = m_token.amount;
     else {
-        amount = (m_token.amount * this_balance.amount) / (vinfo.supply.amount + m_token.amount);
+        amount = static_cast<int64_t>((static_cast<uint128_t>(m_token.amount) * static_cast<uint128_t>(this_balance.amount))
+                                      / (static_cast<uint128_t>(vinfo.supply.amount) + static_cast<uint128_t>(m_token.amount)));
```

Comment:
too many casts )
first two should be enough.
@kaynarov, your original implementation is similar, am I missing something?